deepchecks/recommender/checks/product_association.py

```diff
@@ -0,0 +1,165 @@
+
+        # Create a new DataFrame to count the non-directional co-occurrence of each pair
+        association_counts = products_association_df.groupby(['item_combination']).size().reset_index(name='co-occurence')
+        # Use the apply method to unpack the tuples into separate columns
+        association_counts[f'prev_{item_col}'], association_counts[item_col] = zip(*association_counts['item_combination'])
+        # Drop the original 'tuple_column' 
+        association_counts.drop(columns=['item_combination'], inplace=True)
+
+        prob_df = filtered_interactions.groupby(item_col).size().reset_index(name='popularity')
+        prob_df['probability'] = prob_df['popularity']/len(filtered_interactions)
+        prob_dict = prob_df.set_index(item_col)['probability'].to_dict()
+        
+        # Function to calculate probability of an item
+        def get_probability(item):
+            return prob_dict.get(item, 0)
+
+        # Add popularity columns to the DataFrame
+        association_counts[f'prev_{item_col}_prb'] = association_counts[f'prev_{item_col}'].apply(get_probability)
+        association_counts[f'{item_col}_prb'] = association_counts[item_col].apply(get_probability)
+        association_counts['lift_metric (%)'] = association_counts['co-occurence']/(association_counts[f'prev_{item_col}_prb']*association_counts[f'{item_col}_prb']*len(filtered_interactions)).astype(np.float16)
+        
+        association_counts = association_counts[[f'prev_{item_col}', item_col,'co-occurence','lift_metric (%)']]
+        movie_translation = item_df[[item_col,item_column_name]].set_index(item_col)[item_column_name].to_dict()
+        
+        association_counts[f'prev_{item_col}']= association_counts[f'prev_{item_col}'].map(movie_translation)
+        association_counts[item_col]= association_counts[item_col].map(movie_translation)
+
+        association_counts = association_counts.sort_values(['co-occurence','lift_metric (%)'],ascending=[False,False]).rename({f'prev_{item_col}':'reference',
+        item_col:'possible recommandation'},axis=1)
+        display_df = association_counts.head(20)
+        # Create a network graph
+        G = nx.Graph()
+
+        # Add edges to the graph based on the lift metric above the threshold
+        for _, row in display_df.iterrows():
+                G.add_edge(row['reference'], row['possible recommandation'], weight=row['lift_metric (%)'])
+
+        # Set node positions using a spring layout
+        pos = nx.spring_layout(G,k=1.2,seed=32)
+
+
+        # Create Plotly nodes
+        nodes_trace = go.Scatter(
+            x=[pos[node][0] for node in G.nodes()],
+            y=[pos[node][1] for node in G.nodes()],
+            mode='markers+text',
+            hovertext=list(G.nodes()),
+            textposition='top center',
+            hoverinfo='text',
+            marker=dict(
+                showscale=False,
+                color=list(dict(G.degree).values()),
```

Comment:
Need legend for these colors

Reply:
fixed:
https://github.com/deepchecks/deepchecks/commit/fa0e410dce9cd742db9fb857414ac3af4676873e